plugin/advisor/catalog/walk_through.go

```diff
@@ -112,11 +114,45 @@ func (d *databaseState) changeState(in tidbast.StmtNode) error {
 	switch node := in.(type) {
 	case *tidbast.CreateTableStmt:
 		return d.createTable(node)
+	case *tidbast.DropTableStmt:
+		return d.dropTable(node)
 	default:
 		return nil
 	}
 }
 
+func (d *databaseState) dropTable(node *tidbast.DropTableStmt) error {
+	// TODO(rebelice): deal with DROP VIEW statement.
+	if !node.IsView {
+		for _, name := range node.Tables {
+			if name.Schema.O != "" && d.name != name.Schema.O {
+				return &WalkThroughError{
+					Type:    ErrorTypeAccessOtherDatabase,
+					Content: fmt.Sprintf("Database `%s` is not the current database `%s`", name.Schema.O, d.name),
+				}
+			}
+
+			schema, exists := d.schemaSet[""]
+			if !exists {
+				schema = d.createSchema("")
+			}
+
+			if _, exists = schema.tableSet[name.Name.O]; !exists {
+				if node.IfExists || !schema.context.CheckCatalog {
```

Comment:
This logic seems to reveal an issue.

The schema records 2 different catalog info

1. The existing catalog info (base)
2. The new catalog info as we walkthrough the new DDL (diff)

Currently, our xxxState doesn't differentiate them. Thus the CheckCatalog logic here is not accurate, because it should only apply to the existing catalog info

It's better to separate `diff` from `base` to make the code more readable. e.g. We can assert the `diff` against a particular DDL.